apps/web/src/mailer/provider/mailgun.ts

```diff
@@ -0,0 +1,52 @@
+import env from '$/env'
+import { type MailerOptions } from '$/mailer/provider'
+import nodemailer from 'nodemailer'
+import mg from 'nodemailer-mailgun-transport'
+import MailMessage from 'nodemailer/lib/mailer/mail-message'
+import SMTPTransport from 'nodemailer/lib/smtp-transport'
+
+type CallbackFn = (
+  err: Error | null,
+  info: SMTPTransport.SentMessageInfo,
+) => void
+
+export default function createMailgunTransport({
+  mailOptions,
+  transportOptions,
+}: MailerOptions) {
+  const domain = env.MAILGUN_DOMAIN
+  const apiKey = env.MAILGUN_API_KEY
+
+  if (!domain || !apiKey) return null
+
+  const transport = mg({ auth: { domain, apiKey } })
+  transport.from = mailOptions.from
+  const transporter = nodemailer.createTransport<SMTPTransport.SentMessageInfo>(
+    transport,
+    transportOptions,
+  )
+  const nonPromiseSendMail = transporter.transporter.send
+  // HACK .send. This has to return a Promise, but it doesn't.
+  const promisifiedSendMail = (
```

Comment:
This is not working. If instead of this I set `transport` in NextAuth -> NodeMailer it works but when sending it fails here because `nodemailer-mailgun-transport` is not returning a Promise I think here:
https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/nodemailer.ts#L75